### PR TITLE
THRIFT-5744: Switch to slog for go library

### PR DESCRIPTION
**Pull request by fishy**
Client: go

### CODE REVIEW

File: lib/go/thrift/simple_server.go [hunk 1]
```diff
@@ -179,8 +179,18 @@ func (p *TSimpleServer) SetForwardHeaders(headers []string) {
...
+// Deprecated: The logging inside TSimpleServer is now done via slog on error
+// level, this does nothing now. It will be removed in a future version.
+func (p *TSimpleServer) SetLogger(_ Logger) {}
+
+// SetLogContext sets the context to be used when logging errors inside
+// TSimpleServer.
+//
+// If this is not called before calling Serve, context.Background() will be
+// used.
+func (p *TSimpleServer) SetLogContext(ctx context.Context) {
```
Comment (fishy) [inline lib/go/thrift/simple_server.go:192]: note: I used the term `LogContext` over `BaseContext` here as I intend this context to only be used for logging.

if we use it as base context to create ctx that we pass into processors and eventually handlers, then whatever developers set on the ctx to control the logging of TSimpleServer will also end up in their service handlers, which is most likely not intended and could be a big surprise.

Comment (dcelasun) [reply lib/go/thrift/simple_server.go:192]: Sounds reasonable.

**Review by dcelasun — APPROVED** [review]

**Comment by fishy** [comment]
the go 1.22 github actions is not ready yet, will rerun github actions once that's ready.

**Comment by fishy** [comment]
> the go 1.22 github actions is not ready yet, will rerun github actions once that's ready.

oh nevermind the new `setup-go` can just download it from upstream if it's not pre-built. neat.

**Pull request merged**